### DISCUSSION TITLE
Dewetron Configuration file (*.dms)

### DISCUSSION THREAD

**Discussion by sreevanii** [DEWETRON/pyDmdReader#8] [General]
Sir, How to read *.dms file in python?
how can I get all channels and their setup, calibration in excel or any other format so that I can access through Python

**Comment by matthiasstraka** [comment]
Hi,
currently there is no python support for reading DMS files that goes beyong reading the raw XML data structure and interpreting it yourself.
To access the DMS part of a DMD file, you can use the following properties:
```python
dmd_file = pyDmdReader.DmdReader(filename)
dmd_file.configuration_xml # gives you the raw XML as a string
dmd_file.configuration_etree # gives you the parses xml tree object
```
For channel configurations, look at the `Channels` element which holds all `Channel` items with all properties.
